src/os/bluestore/BlueStore.cc

```diff
@@ -10801,7 +10812,10 @@ int BlueStore::_do_clone_range(
       // make sure it is shared
       if (!blob.is_shared()) {
 	c->make_blob_shared(_assign_blobid(txc), e.blob);
-	dirtied_oldo = true;  // fixme: overkill
+	if (dirty_range_begin == 0) {
+          dirty_range_begin = e.logical_offset;
+        } 
+        dirty_range_end = e.logical_end();
```

Comment:
should be `dirty_range_end = e.logical_end() - 1;` here to exclude next potential shard